src/Symfony/Bridge/Doctrine/Security/User/EntityUserProvider.php

```diff
@@ -116,8 +117,12 @@ public function supportsClass(string $class)
     /**
      * {@inheritdoc}
      */
-    public function upgradePassword(UserInterface $user, string $newEncodedPassword): void
+    public function upgradePassword($user, string $newEncodedPassword): void
```

Comment:
This class is not final, so this change is a BC break.

Reply:
Good catch, type change reverted and `@final` annotation added so that we can change the signature when adding back the method to the `PasswordUpgraderInterface` in 6.0